src/cdk-experimental/scrolling/virtual-scroll-viewport.ts

```diff
@@ -174,25 +162,12 @@ export class CdkVirtualScrollViewport implements OnInit, DoCheck, OnDestroy {
     Promise.resolve().then(() => {
       this._viewportSize = this.orientation === 'horizontal' ?
           this.elementRef.nativeElement.clientWidth : this.elementRef.nativeElement.clientHeight;
-      this._ngZone.runOutsideAngular(() => {
-        fromEvent(this.elementRef.nativeElement, 'scroll').subscribe(() => {
-          this._markScrolled();
-        });
-      });
+      fromEvent(this.elementRef.nativeElement, 'scroll').pipe(throttleTime(0, animationFrame))
```

Comment:
`requestAnimationFrame` runs inside the `NgZone` as well.

Reply:
I think this should be ok though. What I was worried about before was that it would trigger change detection for every `scroll` event, now its throttled by `requestAnimationFrame`. Even if I subscribe outside the zone, I'm just going to have to go back into the zone and mark for change detection right?

Reply:
I'm not familiar with the rxjs internals, but even if it is throttled by `requestAnimationFrame`, wouldn't it still fire for more or less every `scroll` event anyway? For the overlay scroll strategies [we run everything outside the zone](https://github.com/angular/material2/blob/master/src/cdk/scrolling/viewport-ruler.ts#L37) and then its up to the strategy [to decide when to re-enter it ](https://github.com/angular/material2/blob/master/src/cdk/overlay/scroll/close-scroll-strategy.ts#L60). Does the virtual scrolling strategy have to re-enter for every pixel or is it only when it swaps out tile? If its the latter, then it should be fine to leave it inside the `NgZone` permanently.

Reply:
Hmm that's a good point, let me see if I can rework this. I at least want to get rid of that `_markScrolled` method